src/modules/PortAudio/PortAudioWriter.cpp

```diff
@@ -203,12 +240,36 @@ bool PortAudioWriter::open()
 
 bool PortAudioWriter::openStream()
 {
+#ifdef Q_OS_MACOS
+	bool bitPerfectMode = sets().getBool("BitPerfect");
+	if (bitPerfectMode && (outputDevice.isEmpty() || outputDevice == "Default"))
+	{
+		int chn = getParam("chn").toInt();
+		outputParameters.device = PortAudioCommon::getDeviceIndexForOutput(outputDevice, chn);
```

Comment:
This is done in `PortAudioWriter::processParams`, soon before calling this function, IMHO not needed here.

Reply:
Are you certain it is always being done and that the setting will never be changed? (Seems unlikely but can't shake the feeling I put the block there for a reason).

I'll re-test.